src/pybind/mgr/dashboard/controllers/rbd_mirroring.py

```diff
@@ -407,6 +407,54 @@ def __call__(self):
                 'content_data': content_data}
 
 
+@APIRouter('/block/mirroring/image/{image_name}', Scope.RBD_MIRRORING)
+@APIDoc("RBD Mirroring Image Management API", "RbdMirroringImage")
+class RbdMirroringImage(RESTController):
+
+    RESOURCE_ID = "image_name"
+
+    MIRROR_MODES = {
+        'journal': rbd.RBD_MIRROR_IMAGE_MODE_JOURNAL,
+        'snapshot': rbd.RBD_MIRROR_IMAGE_MODE_SNAPSHOT
+    }
+
+    @Endpoint(method='POST')
+    @UpdatePermission
+    @handle_rbd_mirror_error()
+    def enable(self, image_name: str, pool_name: str, mode: str, namespace: str = ''):
+        res = RbdMirroringService.enable_image(
+            image_name, pool_name, namespace, self.MIRROR_MODES[mode])
+        return res
+
+    @Endpoint(method='POST')
+    @UpdatePermission
+    @handle_rbd_mirror_error()
+    def disable(self, image_name: str, pool_name: str, namespace: str = ''):
+        res = RbdMirroringService.disable_image(image_name, pool_name, namespace)
+        return res
+
+    @Endpoint(method='POST')
+    @UpdatePermission
+    @handle_rbd_mirror_error()
+    def promote(self, image_name: str, pool_name: str, namespace: str = '', force: bool = False):
+        res = RbdMirroringService.promote_image(image_name, pool_name, namespace, force)
+        return res
+
+    @Endpoint(method='POST')
+    @UpdatePermission
+    @handle_rbd_mirror_error()
+    def demote(self, image_name: str, pool_name: str, namespace: str = ''):
+        res = RbdMirroringService.demote_image(image_name, pool_name, namespace)
+        return res
```

Comment:
Mirrored images have an attribute that designates the image as primary or non-primary. Promoting changes primary to non-primary and demoting from non-primary to primary. I guess we should display whether an image is primary if we enable mirroring.

Reply:
This is then the best example for the declarative vs. imperative approach. Now we have a new RBD Image property called `primary` (or `primary_mirror`), so if a user wants to change that, instead of calling a promote/demote endpoint, they should update the RBD Image status with `primary: true` or `false`, and internally we would call promote/demote accordingly.

Reply:
Done. I still think that going this way is a bit inconsistent with what we are already doing. Usually we perform actions to change the attributes, but this time we edit the image to perform an rbd action internally.